src/cloudflare/internal/vectorize-api.ts

```diff
@@ -166,7 +167,8 @@ class VectorizeIndexImpl implements Vectorize {
       headers: {
         'content-type': contentType,
         'cf-vector-search-dim-width': String(
-          vectors.length ? vectors[0]?.values?.length : 0
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+          vectors.length ? vectors.at(0)?.values?.length : 0
```

Comment:
Why does this need changed?

Reply:
it doesn't had to be changed, i just had to add a eslint disable next line there. somehow there is a bug, and with or without [0]/.at(0), it still fails. 

This will probably need to be done in a separate PR but we need to disallow using unsafe array index in our tsconfig, since array[123123] will not resolve into T | undefined, it will resolve to T, unles we use .at(123123)